plugin/code_lens.py

```diff
@@ -131,10 +133,36 @@ def resolved_code_lens(self) -> Iterable[CodeLensData]:
             if lens.is_resolved():
                 yield lens
 
-    def render(self) -> None:
-        accent = self.view.style_for_scope("region.greenish markup.accent.codelens.lsp")["foreground"]
-        for index, lens in enumerate(self._flat_iteration()):
-            self.view.add_regions(self._region_key(index), [lens.region], "", "", 0, [lens.small_html], accent)
+    def _get_phantom_region(self, region: sublime.Region) -> sublime.Region:
+        """Takes the current region and puts the phantom on the line above, properly indented.
+
+        Code-lenses are typically shown above the code in VSCode.
+        """
+        line = self.view.substr(self.view.line(region))
+        offset = 0
+        for ch in line:
+            if ch.isspace():
+                offset += 1
+            else:
+                break
+
+        row, _ = self.view.rowcol_utf16(region.a)
+        point = self.view.text_point_utf16(max(row - 1, 0), offset, clamp_column=True)
+        return sublime.Region(point)
+
+    def render(self, mode: str) -> None:
+        if mode == 'annotation':
+            accent = self.view.style_for_scope("region.greenish markup.accent.codelens.lsp")["foreground"]
+            for index, lens in enumerate(self._flat_iteration()):
```

Comment:
Just a nit but that `_flat_iteration` method seems kinda redundant. It's an extra 3-lines method with an extra, redundant iteration loop when you can just use `self._code_lenses.values()` in place of it.

Reply:
It's a helper method because I repeatedly call it. If flattening it in-line is better is considered better then that's fine. I'm not sure how it's redundant though. It's taking a nested structure and flattening it into a single iteration instead of two.

Reply:
Note that this is equivalent to: 
```py
for group in self._code_lenses.values():
   for lens in group:
       ...
```
The helper method is using `yield from` not `yield`.

Reply:
Right. I'm wrong here. Misunderstood the logic